velociraptor/observations/objects.py

```diff
@@ -246,50 +254,72 @@ def load(self, filename: str):
         filename: str
             The filename to load the data from. Probably should end in
             .hdf5.
+
+        prefix: str, optional
+            An optional prefix to all fields that enables multiple
+            observational datasets to be saved in a single file. Not
+            for general use, only within :class:`MultiRedshiftObservationalData`.
+            If a prefix is used, cosmology is not saved.
+
         """
 
+        if prefix is not None:
+            # To enable human-readable files
+            prefix = f"{prefix}_"
+        else:
+            prefix = ""
+
         self.filename = filename
 
         # Load data here.
-        self.x = unyt_array.from_hdf5(filename, dataset_name="values", group_name="x")
-        self.y = unyt_array.from_hdf5(filename, dataset_name="values", group_name="y")
+        self.x = unyt_array.from_hdf5(
+            filename, dataset_name=f"{prefix}values", group_name="x"
+        )
+        self.y = unyt_array.from_hdf5(
+            filename, dataset_name=f"{prefix}values", group_name="y"
+        )
         self.x_units = self.x.units
         self.y_units = self.y.units
 
         try:
             self.x_scatter = unyt_array.from_hdf5(
-                filename, dataset_name="scatter", group_name="x"
+                filename, dataset_name=f"{prefix}scatter", group_name="x"
             )
         except KeyError:
             self.x_scatter = None
 
         try:
             self.y_scatter = unyt_array.from_hdf5(
-                filename, dataset_name="scatter", group_name="y"
+                filename, dataset_name=f"{prefix}scatter", group_name="y"
             )
         except KeyError:
             self.y_scatter = None
 
         with h5py.File(filename, "r") as handle:
-            metadata = handle["metadata"].attrs
+            metadata = handle[f"{prefix}metadata"].attrs
 
             self.comment = metadata["comment"]
             self.name = metadata["name"]
             self.citation = metadata["citation"]
             self.bibcode = metadata["bibcode"]
             self.redshift = metadata["redshift"]
+            self.redshift_lower = metadata["redshift_lower"]
+            self.redshift_upper = metadata["redshift_upper"]
```

Comment:
Don't we want backward compatibility (for `.hdf5` files that were created in the past and hence do not carry these two new attributes)? 

```suggestion
            self.redshift_lower = metadata.get("redshift_lower", self.redshift)
            self.redshift_upper = metadata.get("redshift_upper", self.redshift)
```